explorer/interpreter/value.h

```diff
@@ -935,6 +935,25 @@ struct ImplConstraint {
   Nonnull<const InterfaceType*> interface;
 };
 
+// A constraint that requires an intrinsic property of a type.
+struct IntrinsicConstraint {
+  // The type that is required to satisfy the intrinsic property.
+  Nonnull<const Value*> type;
+  // The kind of the intrinsic property.
+  enum Kind {
+    // `type` intrinsically implicitly converts to `parameters[0]`.
+    // TODO: Split this up into finer-grained checks.
+    ImplicitAs,
+  };
+  Kind kind;
+  // Arguments for the intrinsic property. The meaning of these depends on
+  // `kind`.
+  std::vector<Nonnull<const Value*>> arguments;
+
+  // Print the intrinsic constraint.
+  void Print(llvm::raw_ostream& out) const;
```

Comment:
This is changing in the next version of the Google style guide. Specifically, for a struct, because the data members are the primary interface, they are permitted to precede member functions instead of appearing at the end of the class.

Reply:
Sure, done, also sent out #2535 suggesting a change to the style rule. But I think we shouldn't block this on that.